x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/GrokPatternCreator.java

```diff
@@ -293,14 +293,18 @@ public String createGrokPatternFromExamples(String seedPatternName, Map<String,
 
         overallGrokPatternBuilder.setLength(0);
 
-        GrokPatternCandidate seedCandidate = new PrecalculatedMappingGrokPatternCandidate(
-            seedPatternName,
-            seedMapping,
-            seedFieldName,
-            grokPatternDefinitions
-        );
+        if (seedPatternName == null) {
+            appendBestGrokMatchForStrings(false, sampleMessages, false, 0);
```

Comment:
Shouldn't the first argument be `true` here?  Because the samples do extend to the end of each message.